skrub/_expressions/_choosing.py

```diff
@@ -424,34 +438,93 @@ class Optional(Choice):
     """A choice between something and nothing."""
 
     def __repr__(self):
+        if self.outcomes[0] is not None or self.outcomes[1] is None:
+            # note when `value` is None, `default` makes no difference
+            value, default = self.outcomes[0], OPTIONAL_VALUE
+        else:
+            value, default = self.outcomes[1], None
         args = _utils.repr_args(
-            (get_default(self),), {"name": self.name}, defaults={"name": None}
+            (value,),
+            {"name": self.name, "default": default},
+            defaults={"name": None, "default": OPTIONAL_VALUE},
         )
         return f"optional({args})"
 
 
-def optional(value, *, name=None):
-    """Construct a choice between a value and ``None``.
+def optional(value, *, name=None, default=OPTIONAL_VALUE):
```

Comment:
note: this is displayed as

![screenshot_2025-05-06T15:01:09+02:00](https://github.com/user-attachments/assets/d9ee57d4-5e29-45a7-a94c-4ff17dae10ba)